src/typeEvaluator/typeEvaluate.ts

```diff
@@ -866,28 +977,38 @@ function handleParentNode({n}: ParentNode, scope: Scope): TypeNode {
 
 function handleNotNode(node: NotNode, scope: Scope): TypeNode {
   const base = walk({node: node.base, scope})
-  if (base.type === 'boolean' && base.value !== undefined) {
-    return {type: 'boolean', value: base.value === false}
-  }
-  return {type: 'boolean'}
+  return mapConcrete(base, scope, (base) => {
```

Comment:
(Related to my feedback on `mapUnion`/`mapConcrete`: If we make this `unknown`-aware we can actually return `nullUnion({type: 'boolean'})` in this case)